src/test.ts

```diff
@@ -136,12 +136,17 @@ export function fileTests(file: string, fileName: string, mayIgnore = defaultIgn
   for (;;) {
     let m = caseExpr.exec(file)
     if (!m) throw new Error("Unexpected file format in " + fileName)
+
+    let [ name, configStr ] = m[1].split(/[{}]/);
```

Comment:
Could you do something like `/(.*?)(\{.*?\})?/.exec` instead here, so that the intention is a bit clearer? The current code will also happily proceed with input like `# Foo {bar: 2` and `# Foo } baz: 2 { } {}` etc.

Reply:
The regex won't work, unfortunately due to the greedy matching semantics of regex parsing.

I did update the parsing via https://github.com/lezer-parser/lezer-generator/pull/3/commits/e0847ec57a6f8b93a6a4277bf7316ca1a5cd2d76.

Hope that brings the required clarity. The method is now robust enough to potentially handle nested configuration keys, too.

Reply:
There are `?`s after the `*`s to make them non-greedy, but indeed, it didn't work. Adding a `$` to the end fixes that. So I still prefer that one-liner to a 13-line parsing function.

Reply:
Updated PR + force pushed. Now works with a one-liner, again.